subsys/usb/class/netusb/function_rndis.c

```diff
@@ -0,0 +1,1000 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_USB_DEVICE_LEVEL
+#define SYS_LOG_DOMAIN "function/rndis"
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+
+#include <usb_device.h>
+#include <usb_common.h>
+
+#include <net/ethernet.h>
+#include <net/net_pkt.h>
+#include <net_private.h>
+
+#include <class/usb_cdc.h>
+
+#include "netusb.h"
+#include "function_rndis.h"
+
+/* RNDIS handling */
+#define CONFIG_RNDIS_TX_BUF_COUNT	5
+#define CONFIG_RNDIS_TX_BUF_SIZE	512
+NET_BUF_POOL_DEFINE(rndis_tx_pool, CONFIG_RNDIS_TX_BUF_COUNT,
+		    CONFIG_RNDIS_TX_BUF_SIZE, 0, NULL);
+static struct k_fifo rndis_tx_queue;
+
+/* Serialize RNDIS command queue for later processing */
+#define CONFIG_RNDIS_CMD_BUF_COUNT	2
+#define CONFIG_RNDIS_CMD_BUF_SIZE	512
+NET_BUF_POOL_DEFINE(rndis_cmd_pool, CONFIG_RNDIS_CMD_BUF_COUNT,
+		    CONFIG_RNDIS_CMD_BUF_SIZE, 0, NULL);
+static struct k_fifo rndis_cmd_queue;
+
+struct k_delayed_work notify_work;
+
+/*
+ * Stack for cmd thread
+ */
+static K_THREAD_STACK_DEFINE(cmd_stack, 1024);
+static struct k_thread cmd_thread_data;
+
+static struct __rndis {
+	u8_t mac[6];
+	u32_t net_filter;
+
+	struct net_if *iface;
+	u8_t ep_in;
+	u8_t ep_notify;
+
+	enum {
+		UNINITIALIZED,
+		INITIALIZED,
+	} state;
```

Comment:
Will there be another state? Can this be a `bool initialized` instead?

Reply:
at the moment I have 2 states, maybe there will be more in the future